src/app-pages/map/components/map-details/components/dam-profile-chart/DamProfileChart.js

```diff
@@ -0,0 +1,1225 @@
+    .attr("fill", "#666666")
+    .attr("font-family", "sans-serif")
+    .attr("font-size", "24px")
+    .text(`${value}'`);
+};
+
+const renderDamProfileChart = (data) => {
+  const dpc = d3.select("#dpc-1");
+  const {
+    mode = "lockTurbine", //could be lock or dam or lockTurbine or turbine. Based off hasLock and hasLock
+    hasLock = true,
+    hasTurbine = true,
+    damTop = 1000,
+    damBottom = 500,
+    currentLevel = 800,
+    tailWater = 600,
+    inflow = 100,
+    outflow = 200,
+    sur = 10,
+    text = "Dam",
+    date = 1571174251, // 15 Oct 2019, 16:17:31
+  } = data;
+  // based off of data, will push labels into horizontalLables
+  //'Top of Dam', 'Bottom of Conservation', 'Pumping Not Feasible', 'Streambed', 'Spillway Crest', 'Top of Surcharge', and 'Design Capacity'.
+  const { horizontalLabels  = [
+    { name: "Top of Dam", value: damTop, showLine: true, side: "left"}
+  ]} = data;
+
+  const tickScale = d3.scaleLinear().domain([0, 17]).range([damTop, damBottom]);
+
+  const damScale = d3.scaleLinear().domain([damTop, damBottom]).range([130, 560]);
+
+  // unused for now - belongs in createMiddleGradient func?
+  // const gradientScale = d3
+  //   .scaleOrdinal()
+  //   .domain([0, 20, 40, 60, 80, 100])
+  //   .range([damScale(gradientTop), damScale(gradientBottom)]);
+
+  // const gradientAxis = d3.axisRight(gradientScale);
+
+  //create line on the left
+  dpc
+    .append("g")
+    .attr("class", "leftLine")
+    .append("path")
+    .attr(
+      "d",
+      curvedLine([
+        [0, 100],
+        [220, 100],
+        [243, 130],
+        [243, 150],
+        [243, 560],
+        [243, 560],
+        [243, 560],
+      ])
+    )
+    .attr("stroke", "#B3B3B3")
+    .attr("stroke-width", 7)
+    .attr("fill", "none")
+    .attr("stroke-linejoin", "miter");
+  //create center dam
+  dpc
+    .select("g.leftLine")
+    .append("path")
+    .attr(
+      "d",
+      straightLine([
+        [410, 561],
+        [410, 150],
+        [390, 150],
+        [390, 130],
+        [510, 130],
+        [510, 150],
+        [490, 170],
+        [490, 210],
+        [560, 440],
+        [610, 440],
+        [610, 561],
+      ])
+    )
+    .attr("stroke", "#B3B3B3")
+    .attr("stroke-width", 2)
+    .attr("fill", "#B3B3B3")
+    .attr("stroke-linejoin", "bevel");
+
+  drawWaterLevel(dpc, currentLevel, damScale);
+  createLegend(dpc);
+  createInflowIcon(dpc);
+  drawMountain(dpc);
+
+  if (hasLock) {
+    createLock(dpc);
+    drawBoat(dpc);
+  } else {
+    if (hasTurbine) {
+      noLockTurbine(dpc);
+    } else {
+      noLock(dpc);
+    } 
+  }
+  if (hasTurbine) {
+    createTurbine(dpc);
+  }
+
+  drawTicks(dpc, tickScale);
+  createOutflowIcon(dpc, mode);
+  createSurchargeIcon(dpc, mode);
+  createMiddleGradient(dpc, damScale);
+  drawDashedLines(dpc, horizontalLabels, damTop, damBottom, damScale);
+  setText(dpc, mode, inflow, outflow, sur, tailWater, text, date);
+};
+
+const ChartDamProfile = ({ data }) => {
```

Comment:
We should keep the component and file name consistent. The file name is DamProfileChart but the component is ChartDamProfile

Reply:
Good catch. Pushing up fix.